mlir/lib/Conversion/LLVMCommon/MemRefBuilder.cpp

```diff
@@ -363,9 +363,9 @@ void UnrankedMemRefDescriptor::computeSizes(
   // Initialize shared constants.
   Value one = createIndexAttrConstant(builder, loc, indexType, 1);
   Value two = createIndexAttrConstant(builder, loc, indexType, 2);
-  Value indexSize =
-      createIndexAttrConstant(builder, loc, indexType,
-                              ceilDiv(typeConverter.getIndexTypeBitwidth(), 8));
+  Value indexSize = createIndexAttrConstant(
+      builder, loc, indexType,
+      llvm::divideCeilSigned(typeConverter.getIndexTypeBitwidth(), 8));
```

Comment:
It looks like this used signed division by accident -- both quantities here are unsigned.